ArcGISRuntimeSDKQt_CppSamples/Scenes/ExploreScenesInFlyoverAR/ExploreScenesInFlyoverAR.cpp

```diff
@@ -0,0 +1,116 @@
+// [WriteFile Name=ExploreScenesInFlyoverAR, Category=Scenes]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ExploreScenesInFlyoverAR.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "IntegratedMeshLayer.h"
+
+using namespace Esri::ArcGISRuntime;
+using namespace Esri::ArcGISRuntime::Toolkit;
+
+ExploreScenesInFlyoverAR::ExploreScenesInFlyoverAR(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(Basemap::imagery(this), this))
+{
+  // create a new elevation source from Terrain3D REST service
+  ArcGISTiledElevationSource* elevationSource = new ArcGISTiledElevationSource(
+        QUrl("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"), this);
+
+  // add the elevation source to the scene to display elevation
+  m_scene->baseSurface()->elevationSources()->append(elevationSource);
+
+  // create the integrated mesh layer
+  const QUrl meshLyrUrl("https://www.arcgis.com/home/item.html?id=dbc72b3ebb024c848d89a42fe6387a1b");
+  m_integratedMeshLayer = new IntegratedMeshLayer(meshLyrUrl, this);
+
+  // add the layer to the scene
+  m_scene->operationalLayers()->append(m_integratedMeshLayer);
+
+  connect(m_integratedMeshLayer, &IntegratedMeshLayer::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+    {
+      qDebug() << e.code() << e.message() << " - " << e.additionalMessage();
+      return;
+    }
+
+    // Enable subsurface navigation. This allows you to look at the scene from below.
+    m_scene->baseSurface()->setNavigationConstraint(NavigationConstraint::None);
+
+    // Start with the camera at the center of the mesh layer.
+    m_originCamera = Camera(m_integratedMeshLayer->fullExtent().center().y(), m_integratedMeshLayer->fullExtent().center().x(), 250, 0, 90, 0);
```

Comment:
would be cleaner to get the `m_integratedMeshLayer->fullExtent().center()` once, then access x and y in the Camera c'tor